sdk/core/perf/src/base_test.cpp

```diff
@@ -0,0 +1,198 @@
+  }
+
+  std::unique_ptr<HttpPolicy> Clone() const override
+  {
+    return std::make_unique<ProxyPolicy>(*this);
+  }
+};
+
+} // namespace
+
+namespace Azure { namespace Perf {
+
+  void BaseTest::ConfigureCoreClientOptions(Azure::Core::_internal::ClientOptions* clientOptions)
+  {
+    if (!m_proxy.empty())
+    {
+      clientOptions->PerRetryPolicies.push_back(std::make_unique<ProxyPolicy>(this));
+    }
+  }
+
+  void BaseTest::PostSetUp()
+  {
+
+    if (!m_proxy.empty())
+    {
+      Azure::Core::_internal::ClientOptions clientOp;
+      clientOp.Retry.MaxRetries = 0;
+      std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesOp;
+      std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesRe;
+      Azure::Core::Http::_internal::HttpPipeline pipeline(
+          clientOp, "PerfFw", "na", std::move(policiesRe), std::move(policiesOp));
+      Azure::Core::Context ctx;
+
+      // Send start-record call
+      {
+        Azure::Core::Url startRecordReq(m_proxy);
+        startRecordReq.AppendPath("record");
+        startRecordReq.AppendPath("start");
+        Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, startRecordReq);
+        auto response = pipeline.Send(request, ctx);
+
+        auto const& headers = response->GetHeaders();
+        auto findHeader = std::find_if(
+            headers.begin(),
+            headers.end(),
+            [](std::pair<std::string const&, std::string const&> h) {
+              return h.first == "x-recording-id";
+            });
+        m_recordId = findHeader->second;
+      }
+
+      // play one test to generate a recording
+      this->Run(ctx);
+      // Run twice to alling with how all other SDK perf langs run
+      this->Run(ctx);
+
+      // Stop recording
+      {
+        Azure::Core::Url stopRecordReq(m_proxy);
+        stopRecordReq.AppendPath("record");
+        stopRecordReq.AppendPath("stop");
+        Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, stopRecordReq);
+        request.SetHeader("x-recording-id", m_recordId);
+        pipeline.Send(request, ctx);
+      }
+
+      // Start playback
+      {
+        Azure::Core::Url startPlayback(m_proxy);
+        startPlayback.AppendPath("playback");
+        startPlayback.AppendPath("start");
+        Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, startPlayback);
+        request.SetHeader("x-recording-id", m_recordId);
+        auto response = pipeline.Send(request, ctx);
+
+        auto const& headers = response->GetHeaders();
+        auto findHeader = std::find_if(
+            headers.begin(),
+            headers.end(),
+            [](std::pair<std::string const&, std::string const&> h) {
+              return h.first == "x-recording-id";
+            });
+        m_recordId = findHeader->second;
+        m_isPlayBackMode = true;
+      }
+    }
+  }
+
+  void BaseTest::PreCleanUp()
```

Comment:
What do you mean by PreCleanUp? Maybe it should be called PostTestRun, similar to PostSetup.

Reply:
The name is following the other languages. I think it was chosen by Micke :),